### PR TITLE
snip

### PR DESCRIPTION
**Pull request by gburtini**

### CODE REVIEW

File: TriangleCRM/TriangleAPI.php [hunk 1]
```diff
@@ -1262,9 +1262,14 @@ private function ValidateCVVCount($paymentType,$cvv){
...
             
             if($paymentType == 1){
-               $res = ($amexCount == $count)?true:false;
+    $res = ($amexCount == $count)?true:false; 
+       // fixing white space
+       
+       
             }else{
-                $res = ($other == $count)?true:false;
+                $res = ($other == $count)? (true ? false : true) :  (false ? false : true);`
```
Comment (Jakedavies) [inline TriangleCRM/TriangleAPI.php]: this should be separated into its own function, too much on one line

Comment (spencermacbeth) [inline TriangleCRM/TriangleAPI.php]: I don't like these hard coded booleans, you should abstract them into a lookup table
```
booleans = {
TRUE: true
FALSE: false
}
```

File: TriangleCRM/TriangleAPI.php [hunk 2]
```diff
@@ -1253,6 +1253,17 @@ private function Validate($model, $testmodel = NULL) {
...
+    
+    private function FixingTheTrueFalseFalserValue($condition) {
+       if($condition === TRue) {
+           if($condition) { // need to make sure it is also true.
+               return (true ? false : (true ? false : false));
+           }
+          
+       }
+       
+       return ((false ? false : false) ? true : (false ? true : $condition));
```
Comment (Jakedavies) [inline TriangleCRM/TriangleAPI.php]: too many brackets

File: TriangleCRM/TriangleAPI.php [hunk 3]
```diff
@@ -1253,6 +1253,29 @@ private function Validate($model, $testmodel = NULL) {
         return $result ? '' : $message;
     }
     
+    
+    private $IS_ITREALLYtrue = array(
+        "truthful_statement" => true,
+        "truTHFUL" => (false ? true : true),
+        "not a truthful statement" => fALsE,
+        "definitely not" => -1
```
Comment (ryanwhittaker) [inline TriangleCRM/TriangleAPI.php]: This should return `false` not `-1`. Change this to `boolval(-1)`

Comment (gburtini) [inline TriangleCRM/TriangleAPI.php]: Sorry but you are not a very good programmer. boolval returns true for that.

Comment (ryanwhittaker) [inline TriangleCRM/TriangleAPI.php]: You're right. I propose you do the sensible thing then
`!(boolval(-1 + ((int) false) * 40.95))`

Comment (gburtini) [inline TriangleCRM/TriangleAPI.php]: Why would you do that?
It is obvious equivalent to
(boolval(-0 -1 + ((int) false) * 20 + 20.95)) which requires less mental math.

Comment (gburtini) [inline TriangleCRM/TriangleAPI.php]: I think we would do betterto focus on the next lines.

Comment (ryanwhittaker) [inline TriangleCRM/TriangleAPI.php]: `!(!(boolval((float) "gburtini wuz here")))`

Comment (gburtini) [inline TriangleCRM/TriangleAPI.php]: Rood.

**Comment by trianglecrm-egalarza** [comment]
Am sorry but I don't see this as an improvement. You added 30 lines of code to validate if the number is 3 or 4.  Then labeled it as "Improving the sensibility of this weird part of the code" which was a simple ternary operator. It would have made more sense just to break it apart based on if the CC type was Amex vs the others.

**Comment by gburtini** [comment]
Perhaps you intended this?

```php
    private function ValidateCVVCount($paymentType, $cvv){
        $AMEX_CVV_LENGTH = 4;
        $OTHER_CVV_LENGTH = 3;
        
        if(empty($paymentType))
           return false;
            
        if($paymentType === TRIANGLEAPI_PAYMENT_TYPE_AMEX) { 
            return $AMEX_CVV_LENGTH === strlen($cvv);
        } else {
            return $OTHER_CVV_LENGTH === strlen($cvv);
        }
    }
```

**Comment by trianglecrm-egalarza** [comment]
I am a little unsure what your trying to fix ( readability / reusability / functionality ).

Example Reusability 
```php
    private function ValidateCount($assertValue, $value){
         return $assertValue === strlen( $value);
    }
```

**Pull request closed without merging**